### PR TITLE
Fixing #5254

### LINKED ISSUE #5254: [BUG] Model Evaluation panel: `load_evaluation` runtime error
### Describe the problem

When using the [new model evaluation panel](https://docs.voxel51.com/user_guide/app.html#model-evaluation-panel-sub-new) from FiftyOne v1.1.0  on the vanilla tutorial, things work like a charm.

But, it fails with an `unsupported operand type(s) for +: 'int' and 'NoneType'` runtime error when evaluating detections via this slightly more elaborated [example](https://docs.voxel51.com/tutorials/evaluate_detections.html).

### Code to reproduce issue

The vanilla example works fine:

```python
import fiftyone.zoo as foz
dataset = foz.load_zoo_dataset("quickstart")  # detections pre-computed
results = dataset.evaluate_detections(
    "predictions",
    gt_field="ground_truth",
    eval_key="eval",
)
```

But, this example fails:

```python
import fiftyone as fo
import fiftyone.zoo as foz
from fiftyone import ViewField as F

model = foz.load_zoo_model('faster-rcnn-resnet50-fpn-coco-torch')
dataset = foz.load_zoo_dataset(
    "coco-2017",
    split="validation",
    dataset_name="evaluate-detections-tutorial",
)
dataset.persistent = True

predictions_view = dataset.take(100, seed=51)
predictions_view.apply_model(model, label_field="faster_rcnn")

high_conf_view = predictions_view.filter_labels("faster_rcnn", F("confidence") > 0.75, only_matches=False)
dataset.save_view("high-confidence", high_conf_view)
results = high_conf_view.evaluate_detections(
    "faster_rcnn",
    gt_field="ground_truth",
    eval_key="eval",
    compute_mAP=True,
)
```

Then open the app > Model evaluation tab > Select the `"eval"` evaluation:

Screenshot 2024-12-10 at 19 10 53

With the following stack trace in the console:

```
File "/home/foo/bar/.venv/lib/python3.12/site-packages/fiftyone/operators/executor.py", line 297, in execute_or_delegate_operator
    result = await do_execute_operator(operator, ctx, exhaust=exhaust)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/foo/bar/.venv/lib/python3.12/site-packages/fiftyone/operators/executor.py", line 341, in do_execute_operator
    result = await (
             ^^^^^^^
  File "/home/foo/bar/.venv/lib/python3.12/site-packages/fiftyone/core/utils.py", line 2355, in run_sync_task
    return await loop.run_in_executor(_get_sync_task_executor(), func, *args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/cedric/.rye/py/cpython@3.12.7/lib/python3.12/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/foo/bar/.venv/lib/python3.12/site-packages/fiftyone/operators/panel.py", line 165, in execute
    method(ctx)
  File "/home/foo/bar/.venv/lib/python3.12/site-packages/fiftyone/operators/builtins/panels/model_evaluation/__init__.py", line 300, in load_evaluation
    metrics["tp"], metrics["fp"], metrics["fn"] = self.get_tp_fp_fn(
                                                  ^^^^^^^^^^^^^^^^^^
  File "/home/foo/bar/.venv/lib/python3.12/site-packages/fiftyone/operators/builtins/panels/model_evaluation/__init__.py", line 115, in get_tp_fp_fn
    sum(ctx.dataset.values(tp_key))
TypeError: unsupported operand type(s) for +: 'int' and 'NoneType'
```

This is happening during the call to the `"@voxel51/operators/model_evaluation_panel_builtin"` operator with the `"load_evaluation "` method.

Given the line which [failed](https://github.com/voxel51/fiftyone/blob/b5f4f395f699c28644f72c3351dc56d47bcde0ac/fiftyone/operators/builtins/panels/model_evaluation/__init__.py#L115), I checked that all corresponding values are not `None`:

```
>>> high_conf_view.values("eval_tp")
[2, 1, 2, 5, 1, 6, 3, 3, 0, 3, 0, 4, 17, 0, 0, 0, 8, 19, 5, 2, 13, 15, 8, 3, 3, 2, 7, 1, 2, 1, 2, 2, 1, 11, 5, 2, 8, 3, 1, 1, 3, 1
, 4, 6, 3, 4, 4, 1, 7, 1, 1, 11, 15, 2, 3, 5, 7, 1, 6, 2, 17, 2, 1, 9, 3, 1, 5, 2, 1, 4, 4, 5, 1, 1, 1, 2, 3, 1, 1, 4, 3, 1, 1, 20
, 3, 3, 1, 2, 2, 3, 2, 2, 1, 5, 3, 1, 2, 2, 4, 2]
>>> high_conf_view.values("eval_fp")
[0, 2, 0, 2, 0, 3, 1, 0, 1, 2, 0, 0, 3, 0, 2, 1, 3, 3, 0, 0, 0, 0, 4, 0, 0, 0, 1, 4, 1, 0, 0, 1, 1, 1, 0, 0, 0, 2, 1, 1, 4, 0, 0,
1, 1, 0, 4, 7, 6, 1, 0, 1, 3, 0, 0, 1, 2, 1, 4, 0, 2, 2, 0, 0, 0, 2, 3, 6, 1, 0, 2, 2, 0, 1, 0, 4, 0, 1, 1, 1, 0, 0, 0, 1, 1, 3, 2
, 0, 1, 1, 0, 0, 1, 0, 0, 1, 1, 0, 4, 0]
>>> high_conf_view.values("eval_fn")
[0, 10, 0, 1, 0, 15, 1, 0, 2, 4, 1, 0, 6, 2, 11, 2, 6, 15, 2, 0, 4, 23, 8, 0, 1, 0, 0, 3, 0, 0, 2, 0, 3, 9, 1, 0, 1, 1, 1, 1, 6, 0
, 2, 3, 2, 0, 2, 5, 8, 0, 0, 11, 11, 0, 0, 0, 4, 1, 1, 0, 4, 13, 0, 2, 3, 1, 2, 7, 0, 0, 0, 3, 0, 0, 0, 8, 0, 5, 0, 0, 1, 0, 1, 26
, 6, 1, 0, 1, 1, 1, 1, 0, 1, 6, 0, 1, 0, 0, 6, 1]
```

(since this looks like a `sum([1, None, 3])` kind of problem)

**=> Is it a bug or misuse? If so, please let me know what I am missing here.**

### System information

-   **OS Platform and Distribution** (e.g., Linux Ubuntu 22.04): Ubuntu 24.04 LTS
-   **Python version** (`python --version`): Python 3.12.7
-   **FiftyOne version** (`fiftyone --version`): FiftyOne v1.1.0, Voxel51, Inc.
-   **FiftyOne installed from** (pip or source): pip (via rye)

### Other info/logs

One difference with the vanilla tutorial is that **the evaluation is computed from a view**. (FWIW, `evaluate_classifications`  seems to work fine with views).

### Willingness to contribute

The FiftyOne Community encourages bug fix contributions. Would you or another
member of your organization be willing to contribute a fix for this bug to the
FiftyOne codebase?

-   [ ] Yes. I can contribute a fix for this bug independently
-   [x] Yes. I would be willing to contribute a fix for this bug with guidance
        from the FiftyOne community
-   [ ] No. I cannot contribute a bug fix at this time

### PR DESCRIPTION
**Pull request by brimoor**
Resolves #5254 

```py

import fiftyone as fo
import fiftyone.zoo as foz

dataset = foz.load_zoo_dataset("quickstart")
dataset.limit(10).evaluate_detections("predictions", eval_key="eval")

session = fo.launch_app(dataset)
```

## Summary by CodeRabbit

- **New Features**
	- Enhanced evaluation metrics for binary classification and object detection.
	- Updated display logic for true positives, false positives, and false negatives based on evaluation type.
	- Added new variables to streamline evaluation method handling.

- **Bug Fixes**
	- Improved accuracy of metric displays in the evaluation component.

- **Documentation**
	- Updated information rows to include evaluation method details for better user clarity.

### CODE REVIEW

File: fiftyone/operators/builtins/panels/model_evaluation/__init__.py [hunk 1]
```diff
@@ -104,29 +105,32 @@ def get_avg_confidence(self, per_class_metrics):
...
-            else None
-        )
-        fn_total = (
-            sum(ctx.dataset.values(fn_key))
-            if dataset.has_field(fn_key)
-            else None
-        )
-        return tp_total, fp_total, fn_total
+    def get_tp_fp_fn(self, info, results):
+        # Binary classification
```
Comment (imanjra) [inline fiftyone/operators/builtins/panels/model_evaluation/__init__.py:109]: We show tp, fp, and fn only for object detection. Should we also show it for binary classification in the panel?

Comment (brimoor) [reply fiftyone/operators/builtins/panels/model_evaluation/__init__.py:109]: It is valid to show TP/FP/FN for binary classification, so yes if you could show it in the UX that would be great!

Comment (imanjra) [reply fiftyone/operators/builtins/panels/model_evaluation/__init__.py:116]: added a commit

**Review by imanjra — APPROVED** [review]
LGTM

**Pull request merged**